webapp/src/main/java/com/box/l10n/mojito/monitoring/PrefixHierarchicalNameMapper.java

```diff
@@ -1,45 +1,74 @@
 
-    public PrefixHierarchicalNameMapper(String namePrefix, String tagKeyPrefix, NamingConvention nameConventionOverride) {
-        Preconditions.checkNotNull(namePrefix, tagKeyPrefix, nameConventionOverride);
-        this.namePrefix = namePrefix;
-        this.tagKeyPrefix = tagKeyPrefix;
+    public PrefixHierarchicalNameMapper(String namePrefix,
+                                        String tagKeyPrefix,
+                                        String stripCharacters,
+                                        @Nullable NamingConvention nameConventionOverride) {
+        this.namePrefix = checkNotNull(namePrefix);
+        this.tagKeyPrefix = checkNotNull(tagKeyPrefix);
+        this.stripCharacters = checkNotNull(stripCharacters);
         this.nameConventionOverride = nameConventionOverride;
     }
 
+    public PrefixHierarchicalNameMapper(String namePrefix,
+                                        String tagKeyPrefix,
+                                        NamingConvention nameConventionOverride) {
+        this(namePrefix, tagKeyPrefix, "", nameConventionOverride);
+    }
+
+    public PrefixHierarchicalNameMapper(String namePrefix,
+                                        String tagKeyPrefix) {
+        this(namePrefix, tagKeyPrefix, "", null);
+    }
+
     /**
      * Pretty much a copy of {@link HierarchicalNameMapper.DEFAULT} with prefixes added.
      */
     @Override
     public String toHierarchicalName(Meter.Id id, NamingConvention namingConvention) {
 
-        if (nameConventionOverride != null) {
-            namingConvention = nameConventionOverride;
-        }
+        namingConvention = Optional.ofNullable(nameConventionOverride).orElse(namingConvention);
 
         String prefix = Strings.isNullOrEmpty(namePrefix) ? "" : (namingConvention.name(namePrefix, Meter.Type.OTHER) + ".");
 
-        String hierarchicalName =
-                        prefix +
-                        id.getConventionName(namingConvention) + id.getConventionTags(namingConvention).stream()
-                        .map(t -> "." + tagKeyPrefix + t.getKey() + "." + t.getValue())
-                        .map(nameSegment -> nameSegment.replace(" ", "_"))
-                        .collect(Collectors.joining(""));
+        return prefix +
+                id.getConventionName(namingConvention) +
+                id.getConventionTags(namingConvention)
+                  .stream()
+                  .map(this::formatTag)
+                  .map(nameSegment -> nameSegment.replace(" ", "_"))
+                  .collect(Collectors.joining(""));
+    }
+
+    private String formatTag(Tag tag) {
+        return "." + tagKeyPrefix + stripCharacters(tag.getKey()) + "." + stripCharacters(tag.getValue());
+    }
 
-        return hierarchicalName;
+    private String stripCharacters(String input){
+        return Optional.ofNullable(Strings.emptyToNull(stripCharacters))
+                       .map(pattern -> input.replaceAll("[" + pattern + "]", ""))
```

Comment:
is pattern character list? small UT for this function?

Reply:
Yeah, it's a character list.

I've added some tests here: https://github.com/box/mojito/pull/600/files#diff-a085e48a4ec35529d79a15b1d6000457R70-R99 they all cover this functionality (along with the tests we already had), but I'll add a couple tests cases for it.